test/realm/RolesRegistryFacet/parcelsRolesRegistryFacetTest.ts

```diff
@@ -0,0 +1,413 @@
+/* global describe it before ethers network */
+/* eslint prefer-const: "off" */
+
+//@ts-ignore
+import { ethers , network } from "hardhat";
+import chai from "chai";
+
+import { Contract } from "ethers";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+
+import { maticRealmDiamondAddress } from "../../../scripts/tile/helperFunctions";
+
+import { deployParcelsRolesRegistryFacet } from "./deployTest";
+
+const { expect } = chai;
+
+describe("ParcelRolesRegistryFacet", async () => {
+  let parcelRolesRegistryFacet: Contract;
+  let grantor: SignerWithAddress;
+  let lender: SignerWithAddress;
+  let grantee: SignerWithAddress;
+  let anotherUser: SignerWithAddress;
+  let mockERC721: any;
+  let anotherMockERC721: any;
+  const ONE_DAY = 60 * 60 * 24;
+
+  const realmDiamondAddress = maticRealmDiamondAddress;
+
+  const ROLE_ALCHEMICA_CHANNELING = ethers.utils.keccak256(
+    ethers.utils.toUtf8Bytes("AlchemicaChanneling()")
+  );
+  const ROLE_TEST = ethers.utils.keccak256(
+    ethers.utils.toUtf8Bytes("testRole()")
+  );
+
+  before(async () => {
+    await network.provider.request({
+      method: "hardhat_reset",
+      params: [
+        {
+          forking: {
+            jsonRpcUrl: process.env.MATIC_URL,
+          },
+        },
+      ],
+    });
+    const signers = await ethers.getSigners();
+    grantor = signers[0];
+    grantee = signers[1];
+    lender = signers[2];
+    anotherUser = signers[3];
+
+    // Deploy the mock ERC721 contract
+    const MockERC721 = await ethers.getContractFactory("MockERC721");
+    mockERC721 = await MockERC721.deploy();
+    await mockERC721.deployed();
+
+    const AnotherMockERC721 = await ethers.getContractFactory("MockERC721");
+    anotherMockERC721 = await AnotherMockERC721.deploy();
+    await anotherMockERC721.deployed();
+
+
+    await deployParcelsRolesRegistryFacet(mockERC721.address);
+
+    // Mint a token to the grantor
+    await mockERC721.mint(grantor.address);
+    await mockERC721.mint(grantor.address);
+
+    parcelRolesRegistryFacet = await ethers.getContractAt(
+      "ParcelRolesRegistryFacet",
+      realmDiamondAddress
+    );
+    // Approve the facet contract to manage grantor's tokens
+    await mockERC721
+      .connect(grantor)
+      .approve(parcelRolesRegistryFacet.address, 1);
+
+    await mockERC721
+      .connect(grantor)
+      .approve(parcelRolesRegistryFacet.address, 2);
+
+    // Set role approval for the facet contract
+    await parcelRolesRegistryFacet
+      .connect(grantor)
+      .setRoleApprovalForAll(
+        mockERC721.address,
+        parcelRolesRegistryFacet.address,
+        true
+      );
+  });
+
+  describe("grantRole", async () => {
+    it("should revert when sender is not grantor or approved", async () => {
```

Comment:
Grantor is an outdated term, we have been using `owner` instead. Recommend replacing its usage
```suggestion
    it("should revert when sender is not owner or approved", async () => {
```